### ISSUE TITLE
Spirit-RF1-driver requires .mbedignore -file

### ISSUE THREAD

**Issue by JanneKiiskila** [ARMmbed/stm-spirit1-rf-driver#1]
The Spirit driver should be designed in such a way that it should not require a `.mbedignore`-file even though it would co-exist with some other drivers. 

Launching the normal mbed-os-example-client and compiling for example a K64F with ESP8266 wifi config yields this failure:

```
gb-cam-ci-mbed-gw-build:~/mbed/mbed-os-example-client> mbed compile -m K64F -t GCC_ARM
Building project mbed-os-example-client (K64F, GCC_ARM)
Scan: .
Scan: env
Scan: mbed
Scan: FEATURE_LWIP
Scan: FEATURE_COMMON_PAL
Scan: FEATURE_STORAGE
Compile [  6.0%]: NanostackRfPhySpirit1.cpp
[Fatal Error] NanostackRfPhySpirit1.h@6,28: NanostackRfPhy.h: No such file or directory
[ERROR] In file included from ./easy-connect/stm-spirit1-rf-driver/source/NanostackRfPhySpirit1.cpp:3:0:
./easy-connect/stm-spirit1-rf-driver/stm-spirit1-rf-driver/NanostackRfPhySpirit1.h:6:28: fatal error: NanostackRfPhy.h: No such file or directory
 #include "NanostackRfPhy.h"
                            ^
compilation terminated.

[mbed] ERROR: "/usr/bin/python" returned error code 1.
[mbed] ERROR: Command "/usr/bin/python -u /home/jankii01/mbed/mbed-os-example-client/mbed-os/tools/make.py -t GCC_ARM -m K64F --source . --build ./BUILD/K64F/GCC_ARM" in "/home/jankii01/mbed/mbed-os-example-client"
```

**Comment by betzw** [comment]
This will be the same for the [Atmel ](ARMmbed/atmel-rf-driver)driver. Not sure if you can risolve this generically w/o `.mbedignore`!

**Comment by JanneKiiskila** [comment]
Yes, they have an issue as well. https://github.com/ARMmbed/atmel-rf-driver/issues/59

**Comment by betzw** [comment]
Any ideas how to _fix_ this?

**Comment by JanneKiiskila** [comment]
There's an internal ticket now on the Networking team, I they have a reference for that in the Atmel -driver. My gut feeling, not possible until Mbed OS 5.8 or later.

**Comment by betzw** [comment]
So, may I not care for this issue until then?

**Comment by JanneKiiskila** [comment]
You might likely be forced to that.
@seppotakalo to confirm

**Comment by SeppoTakalo** [comment]
Hi,

So yes, currently all RF drivers implementing the Nanostack's RF interface require `FEATURE_NANOSTACK` to be enabled in Mbed OS.
This is a unfortunate side effect of optimising the build time and we are planning to get rid of it in the future.
I cannot make any commitment whether it will be 5.8 or later.

I therefore recommend closing this issue as it is not something that can be affected from the driver side.

**Comment by JanneKiiskila** [comment]
Raised the same issue also on the [mcr20a](https://github.com/ARMmbed/mcr20a-rf-driver/issues/13) as well. Full set now.

**Comment by SeppoTakalo** [comment]
Fixed in #3

**Comment by betzw** [comment]
@JanneKiiskila, can you pls. close the issue?
